game-core/src/main/java/games/strategy/triplea/delegate/DiceRoll.java

```diff
@@ -110,74 +110,26 @@ public static DiceRoll rollAa(
       final Territory location,
       final CombatValue combatValueCalculator) {
 
-    final GameData data = bridge.getData();
+    final String typeAa = UnitAttachment.get(aaUnits.iterator().next().getType()).getTypeAa();
+    final GamePlayer player = aaUnits.iterator().next().getOwner();
```

Comment:
Would you mind updating the documentation that 'aaUnits' must have at least one element in it?

Reply:
Side-note, perhaps worth doing here as we've re-written the entire method almost, most of the javadocs on the method are not very valuable.

EG:
- " aaUnits - AA units that could potentially be rolling" < does not really add much information. 
- "@param location - battle territory" < seems like the parameter should just be renamed
- "* @param bridge - delegate bridge" < ditto, parameter could just be renamed.

Reply:
I've updated the documentation.  The requirement that 'aaUnits' have at least on element isn't new after this refactor.  It has always been a requirement.  In a future PR, I plan to rework the parameters to this method so that requirement is no longer needed.